pscheduler-archiver-kafka/kafka/archive

```diff
@@ -10,25 +10,115 @@ from kafka.errors import KafkaError
 
 MAX_SCHEMA = 1
 
+log_prefix="archiver-kafka"
+
+log = pscheduler.Log(prefix=log_prefix, quiet=False)
+
+archiver = None
+
+class psKafkaArchiver(object):
+    def is_connected(self):
+        """
+        Determine if producer is connected
+        """
+        return self.producer is not None
+
+    def close(self):
+        """
+        Close producer connection
+        """
+        self.producer.close()
+        self.producer=None
+
+    def publish(self, message):
+        """
+        Send message to producer
+        """
+        if self.is_connected() is True:
+            self.producer.send(self.topic, message)
+
+    def __init__(self, archiver_id, bootstrap_servers,topic, retries=3, security_protocol='',ssl_cafile=None,ssl_certfile=None,ssl_keyfile=None,ssl_password='', ssl_checkhostname=False ):
+        self.bootstrap_servers = bootstrap_servers
+        self.topic=topic
+        self.ssl_checkhostname=False
+        self.client_id=archiver_id
+        self.retries= retries
+        if(security_protocol=='SSL'):
+            self.security_protocol=security_protocol
+            if(ssl_cafile is None or ssl_certfile is None or ssl_keyfile is None):
+                raise ValueError("ssl files missing")
+            self.ssl_cafile = ssl_cafile
```

Comment:
Having the archive spec point to a file poses a security problem because someone remote can ask the archiver to read any file to which it has access.  That doesn't mean the contents will end up anywhere off-system, but there are places in `/dev` where that could have undesirable side effects.  I'm not dead-set against it, but this should be considered carefully to avoid having to do another release that fixes things an auditor doesn't like.

I can think of a couple of alternate ways to go about  this, in order of preference:

 * Provide the certificate and key as strings and have the `psKafkaArchiver` hold the contents in temporary files with permissions of `0600` during the lifetime of the object.  The CA bundle is a thorny point because it can get large (~200K on my systems).  If Kafka's SSL implementation will use the system's bundle by default, that would be a good compromise and still allows providing one.  Any parameters that end up containing sensitive information should be renamed so they begin with an underscore (e.g., `_foo`).  This will prevent the API from revealing their values to outside entities.

 * Allow file names (just names, no absolute or relative paths) but require they exist in some sandboxed directory such as `/etc/pscheduler/archiver/kafka` to be used.  The pitfall here is that it prevents a third party from running a task that archives to their own Kafka server with full SSL.

Reply:
- I have refactored the code so the certs and keys are passed as strings and is held in temp files during the lifetime of the object. 
- Also, the certs and keys have been made private vars so they are not exposed through the api